cfbs/masterfiles/generate_vcf_git_checkout.py

```diff
@@ -0,0 +1,92 @@
+import os
+import shutil
+import subprocess
+import sys
+
+from cfbs.git import git_exists
+from cfbs.utils import write_json
+from cfbs.masterfiles.analyze import (
+    finalize_vcf,
+    initialize_vcf,
+    versions_checksums_files,
+)
+
+DIR_PATH = "."
+"""The path of the working directory."""
+
+MPF_URL = "https://github.com/cfengine/masterfiles"
+MPF_PATH = os.path.join(DIR_PATH, "masterfiles")
+
+
+def generate_vcf_git_checkout(interesting_tags=None):
+    # clone the MPF repo every time the script is run, in case there are updates
+    if os.path.isdir(MPF_PATH):
+        shutil.rmtree(MPF_PATH)
+
+    subprocess.run(
+        ["git", "clone", MPF_URL],
+        cwd=DIR_PATH,
+        check=True,
+    )
+
+    if not git_exists():
+        print("`git` was not found")
+        sys.exit(1)
+
+    result = subprocess.run(
+        ["git", "tag"], cwd=MPF_PATH, capture_output=True, check=True
+    )
+    tags = result.stdout.decode("UTF-8").splitlines()
+
+    # if not given, choose tags to checkout - by default, only consider version releases
+    if interesting_tags is None:
+        interesting_tags = []
+
+        for tag in tags:
+            if "-" not in tag:
+                interesting_tags.append(tag)
+
+    versions_dict, checksums_dict, files_dict = initialize_vcf()
+
+    for tag in interesting_tags:
+        print(tag)
+
+        # checkout the version tag
+        subprocess.run(
+            ["git", "checkout", "--force", tag],
+            cwd=MPF_PATH,
+            check=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        # a clean is necessary to remove all the undesired files
+        subprocess.run(
+            ["git", "clean", "-dffx"],
+            cwd=MPF_PATH,
+            check=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+
+        # build masterfiles from git as they are in the tarball packages
```

Comment:
Seems you should check that autoconf and automake are installed first so that autogen.sh can work.

Reply:
Since you are using `check=True` with `run()` but not catching any exceptions I worry about hard to read stack traces in case of problems. Probably fine in the exceptional cases but something like autoconf missing should be checked I think as that seems likely to happen.

Reply:
Added checking for required commands in https://github.com/cfengine/cfbs/pull/208/commits/b516b19c038d149e10df342e60b71cb777cc34b5.

As for the stack traces, I've checked a few cases and I think it's readable enough. For the git command calls, the failing command and the status code should be enough information, and for the autogen and make calls, currently the entire stdout and stderr are printed.